crates/services/bls_aggregation/src/lib.rs

```diff
@@ -1,7 +1,367 @@
+//! #     let operator_state_retriever_address =
+//! #         get_operator_state_retriever_address(http_endpoint.to_string()).await;
+//! #     
+//! #     // Create avs clients to interact with contracts deployed on anvil
+//! #     let avs_registry_reader = AvsRegistryChainReader::new(
+//! #         get_test_logger(),
+//! #         registry_coordinator_address,
+//! #         operator_state_retriever_address,
+//! #         http_endpoint.to_string(),
+//! #     )
+//! #     .await
+//! #     .unwrap();
+//! #
+//! #     // Create operators info service
+//! #     let operators_info = OperatorInfoServiceInMemory::new(
+//! #         get_test_logger(),
+//! #         avs_registry_reader.clone(),
+//! #         ws_endpoint.to_string(),
+//! #     )
+//! #     .await
+//! #     .unwrap()
+//! #     .0;
+//! #
+//! #     // Create avs registry service chain caller
+//! #     let avs_registry_service =
+//! #         AvsRegistryServiceChainCaller::new(avs_registry_reader.clone(), operators_info);
+//! #    let (service_handle, mut aggregate_receiver) =
+//! #        BlsAggregatorService::new(avs_registry_service, logger).start();
+//! #
+//!     match aggregate_receiver.receive_aggregated_response().await {
+//!         Ok(aggregated_response) => {
+//!             // Handle the aggregated response
+//!         }
+//!         Err(e) => {
+//!             // Handle the error
+//!         }
+//!     }
+//! # }
+//! ```
+//!
+//! ## Example Diagram
+//!
+//! The following diagram shows the sequence of events when a user creates the BLS Aggregator Service, starts the service, and then initializes a task. The service processes two signatures from the operators and then aggregates them into a single aggregated signature.
+//!
+//! <pre>
+#![cfg_attr(doc, doc = simple_mermaid::mermaid!("../diagram/sequence-bls.mmd"))]
+//! </pre>
+//!
+//! ## Testing
+//!
+//! To run the [integration tests](https://github.com/Layr-Labs/eigensdk-rs/blob/dev/crates/services/bls_aggregation/src/bls_agg_test.rs), you can use the following command:
+//!
+//! ```sh
+//! cargo test --package eigen-services-blsaggregation --lib -- bls_agg_test::integration_test --show-output
+//! ```
+//! [`new()`]: bls_agg::BlsAggregatorService::new()
+//! [`start()`]: bls_agg::BlsAggregatorService::start()
+//! [`initialize_task(metadata: TaskMetadata)`]: bls_agg::ServiceHandle::initialize_task()
+//! [`process_signature(task_signature: TaskSignature)`]: bls_agg::ServiceHandle::process_signature()
+//! [`receive_aggregated_response()`]: bls_agg::AggregateReceiver::receive_aggregated_response()
+//! [`with_window_duration()`]: bls_agg::TaskMetadata::with_window_duration()
+//! [`ServiceHandle`]: bls_agg::ServiceHandle
+//! [`AggregateReceiver`]: bls_agg::AggregateReceiver
+//! [`TaskMetadata`]: bls_agg::TaskMetadata
+//! [`TaskSignature`]: bls_agg::TaskSignature
+//! [`BlsAggregatorService`]: bls_agg::BlsAggregatorService
```

Comment:
Just FYI, these can be inlined like: `[new()]` -> `[new()](bls_agg::BlsAggregatorService::new())`